eventdisplay/tracks/FairEveTracks.h

```diff
@@ -0,0 +1,132 @@
+/********************************************************************************
+ *    Copyright (C) 2020 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+/*
+ * FairEveTracks.h
+ *
+ *  Created on: 4 maj 2020
+ *      Author: Daniel Wielanek
+ *		E-mail: daniel.wielanek@gmail.com
+ *		Warsaw University of Technology, Faculty of Physics
+ */
+#ifndef FAIRTEVERACKS_H_
+#define FAIRTEVERACKS_H_
+#include "FairEventManager.h"
+#include "FairTask.h"
+
+#include <Rtypes.h>
+#include <RtypesCore.h>
+#include <TClonesArray.h>
+#include <TEveElement.h>
+#include <TEveTrack.h>
+#include <TEveTrackPropagator.h>
+#include <TGeoTrack.h>
+#include <TNamed.h>
+#include <memory>
+
+/**
+ * base evelement that represents tracks
+ */
+class FairEveTracks
+    : public TEveElement
+    , public TNamed
+{
+    FairEventManager *fEventManager;
+    std::unique_ptr<TObjArray> fEveTrList;
```

Comment:
```suggestion
```

Do we still need this list at all?

See for example:

```diff
diff --git a/eventdisplay/tracks/FairEveTracks.cxx b/eventdisplay/tracks/FairEveTracks.cxx
index c51dd7f41..2902014c1 100644
--- a/eventdisplay/tracks/FairEveTracks.cxx
+++ b/eventdisplay/tracks/FairEveTracks.cxx
@@ -18,6 +18,7 @@
 #include "FairEveRecoTrackList.h"
 #include "FairEveTrack.h"

+#include <algorithm>
 #include <TEveElement.h>
 #include <TEveManager.h>
 #include <TEveSelection.h>
@@ -25,7 +26,6 @@

 FairEveTracks::FairEveTracks(Bool_t acceptCompound)
     : fEventManager(nullptr)
-    , fEveTrList(nullptr)
     , fPt{0, 10}
     , fEta{-10, 10}
     , fEnergy{0, 10}
@@ -33,37 +33,24 @@ FairEveTracks::FairEveTracks(Bool_t acceptCompound)
     , fUseEta(kFALSE)
     , fUseEnergy(kFALSE)
     , fAcceptCompound(acceptCompound)
-{
-    fEveTrList = std::unique_ptr<TObjArray>(new TObjArray());
-}
+{}

 void FairEveTracks::ResetGroup()
 {
-    TEveElement *el = dynamic_cast<TEveElement *>(this);
-    if (el == nullptr)
-        return;
-    for (Int_t iTrackGroup = 0; iTrackGroup < fEveTrList->GetEntriesFast(); iTrackGroup++) {
-        TEveTrackList *ele = static_cast<TEveTrackList *>(fEveTrList->At(iTrackGroup));
-        gEve->RemoveElement(ele, el);
-    }
-    fEveTrList->Clear();
+    RemoveElements();
 }

 void FairEveTracks::ToggleTracks()
 {
-    for (int iTrackGroup = 0; iTrackGroup < fEveTrList->GetEntriesFast(); iTrackGroup++) {
-        TEveTrackList *trackList = (TEveTrackList *)fEveTrList->UncheckedAt(iTrackGroup);
-        for (TEveElement::List_i eveTrackIterator = trackList->BeginChildren();
-             eveTrackIterator != trackList->EndChildren();
-             ++eveTrackIterator) {
-            FairEveTrack *track = dynamic_cast<FairEveTrack *>((*eveTrackIterator));
+    std::for_each(BeginChildren(), EndChildren(), [](TEveElement* trackGroup) {
+        std::for_each(trackGroup->BeginChildren(), trackGroup->EndChildren(), [](TEveElement* track) {
             if (track->GetRnrSelf()) {
                 track->SetRnrSelfChildren(kFALSE, kFALSE);
             } else {
                 track->SetRnrSelfChildren(kTRUE, kTRUE);
             }
-        }
-    }
+        });
+    });
     gEve->Redraw3D(kFALSE);
 }

@@ -77,32 +64,28 @@ FairEveTracks::~FairEveTracks() {}

 TEveTrackList *FairEveTracks::FindTrackGroup(TString groupName, Color_t color)
 {
-    TEveTrackList *TrackGroup = nullptr;
-    TrackGroup = static_cast<TEveTrackList *>(FindChild(groupName));
-    if (TrackGroup == nullptr) {
-        TEveTrackPropagator *TrPropagator = new TEveTrackPropagator();
+    auto trackGroup(static_cast<TEveTrackList *>(FindChild(groupName)));
+    if (trackGroup == nullptr) {
         if (fAcceptCompound) {
-            TrackGroup = new FairEveRecoTrackList(groupName, TrPropagator);
+            trackGroup = new FairEveRecoTrackList(groupName, new TEveTrackPropagator());
         } else {
-            TrackGroup = new TEveTrackList(groupName, TrPropagator);
+            trackGroup = new TEveTrackList(groupName, new TEveTrackPropagator());
         }
-        TrackGroup->SetMainColor(color);
-        fEveTrList->Add(TrackGroup);
-        AddElement(TrackGroup);
-        TrackGroup->SetRnrLine(kTRUE);
+        AddElement(trackGroup); // takes ownership!
+        trackGroup->SetMainColor(color);
+        trackGroup->SetRnrLine(kTRUE);
     }
-    return TrackGroup;
+    return trackGroup;
 }

 void FairEveTracks::ToggleGroups()
 {
-    for (int iTrackGroup = 0; iTrackGroup < fEveTrList->GetEntriesFast(); iTrackGroup++) {
-        TEveTrackList *trackGroup = (TEveTrackList *)fEveTrList->UncheckedAt(iTrackGroup);
+    std::for_each(BeginChildren(), EndChildren(), [](TEveElement* trackGroup) {
         if (trackGroup->GetRnrSelf()) {
             trackGroup->SetRnrSelfChildren(kFALSE, kFALSE);
         } else {
             trackGroup->SetRnrSelfChildren(kTRUE, kTRUE);
         }
-    }
+    });
     gEve->Redraw3D(kFALSE);
 }
diff --git a/eventdisplay/tracks/FairEveTracks.h b/eventdisplay/tracks/FairEveTracks.h
index 4385b370f..19c921ae1 100644
--- a/eventdisplay/tracks/FairEveTracks.h
+++ b/eventdisplay/tracks/FairEveTracks.h
@@ -36,7 +36,6 @@ class FairEveTracks
     , public TNamed
 {
     FairEventManager *fEventManager;
-    std::unique_ptr<TObjArray> fEveTrList;
     Double_t fPt[2];
     Double_t fEta[2];
     Double_t fEnergy[2];
@@ -52,7 +51,6 @@ class FairEveTracks
     Double_t GetEtaMax() const { return fEta[1]; }
     Double_t GetEnergyMin() const { return fEnergy[0]; };
     Double_t GetEnergyMax() const { return fEnergy[1]; };
-    ;
     Bool_t UsePtCut() const { return fUsePt; };
     Bool_t UseEtaCut() const { return fUseEta; };
     Bool_t UseEnergyCut() const { return fUseEnergy; };
@@ -61,7 +59,6 @@ class FairEveTracks
      */
     void ResetGroup();
     FairEventManager *GetEventManager() const { return fEventManager; };
-    TObjArray *GetTracksList() const { return fEveTrList.get(); };
     /**
      *
      * @param groupName - name of the track group
diff --git a/examples/advanced/propagator/src/FairEveRecoTracksExample.cxx b/examples/advanced/propagator/src/FairEveRecoTracksExample.cxx
index 3f61c4b52..12962b984 100644
--- a/examples/advanced/propagator/src/FairEveRecoTracksExample.cxx
+++ b/examples/advanced/propagator/src/FairEveRecoTracksExample.cxx
@@ -210,20 +210,4 @@ void FairEveRecoTracksExample::SetDrawMC(Bool_t draw)
     }
 }

-void FairEveRecoTracksExample::ToggleTracks()
-{
-    for (int i = 0; i < GetTracksList()->GetEntriesFast(); i++) {
-        TEveTrackList *eveList = (TEveTrackList *)GetTracksList()->UncheckedAt(i);
-        for (TEveElement::List_i iter = eveList->BeginChildren(); iter != eveList->EndChildren(); ++iter) {
-            FairEveRecoTrack *el = dynamic_cast<FairEveRecoTrack *>((*iter));
-            if (el->GetRnrSelf()) {
-                el->SetRnrSelfChildren(kFALSE, kFALSE);
-            } else {
-                el->SetRnrSelfChildren(kTRUE, kTRUE);
-            }
-        }
-    }
-    gEve->Redraw3D(kFALSE);
-}
-
 FairEveRecoTracksExample::~FairEveRecoTracksExample() {}
diff --git a/examples/advanced/propagator/src/FairEveRecoTracksExample.h b/examples/advanced/propagator/src/FairEveRecoTracksExample.h
index 376f07dde..b14476350 100644
--- a/examples/advanced/propagator/src/FairEveRecoTracksExample.h
+++ b/examples/advanced/propagator/src/FairEveRecoTracksExample.h
@@ -55,7 +55,6 @@ class FairEveRecoTracksExample : public FairEveTracks
         fShowPrimary = prim;
         fShowSecondary = sec;
     }
-    virtual void ToggleTracks();
     virtual InitStatus Init();
     virtual ~FairEveRecoTracksExample();
     ClassDef(FairEveRecoTracksExample, 0)
```

Reply:
I fixed headers with eclipse-cdt. I hope now it is fine. I also commented out some objects that should not be streamed.